server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -137,6 +137,18 @@ public class RemoteClusterStateService implements Closeable {
         Setting.Property.NodeScope
     );
 
+    /**
+     * If enabled, the remote publication flow will always apply full state and not use the diff.
+     * This is to be used as an andon cord in case we want to rebuild the full cluster state on nodes from remote.
+     * If disabled, we will use the diff manifest shared by cluster manager to build the cluster state on nodes.
+     */
+    public static final Setting<Boolean> REMOTE_PUBLICATION_APPLY_FULL_STATE = Setting.boolSetting(
```

Comment:
Disabling the remote state will require us to change the setting in `opensearch.yml` file and nodes to be bounced, where as this setting provides us an option to dynamically disable the diff publication.
This setting is added specifically for the case if we identify any issue with diff publication or application which causes the cluster state to go stale, we can toggle this setting so that nodes can sync the cluster state to latest state with master.
For dealing with issues like throttling while downloading a huge cluster state, we can add some jitter while downloading the state.

Reply:
It only requires cluster manager nodes to be restarted and that can be done in rolling restart fashion as well. If we adding this new dynamic setting, I am thinking then why not make remote publication as dynamic setting itself.

Reply:
Yes, I don't have any argument of not keeping this setting as `Dynamic`, but the way currently the publication flow has been written just turning the setting to Dynamic won't work. We will need to re-access the whole publication flow as it was written on assumption of the setting being `Final`.

Reply:
In case where we get into a state of diffs corrupting the cluster state a good way would be to ensure we do a full download. Rolling restart for masters might be a temporary arrangement to have data nodes download the full state?

Reply:
Remote publication is currently gated using the static feature flag for experimental release. This has to be changed to a new dynamic setting for gating in GA release. This is a more important andon cord IMO.  
The remote state setting is final setting and it only impacts the uploads. So this should be fine for now.

Reply:
i don't think we should introduce more settings for the sake of it. If there is bug in publication flow, you disable publication flow. why do you need another andon cord?

Reply:
@shwetathareja I think we are talking about the same thing - Remote Publication setting. We need to make it dynamic which is not the case currently.